invenio_github/cli.py

```diff
@@ -0,0 +1,312 @@
+        if user is None:
+            raise Exception("User {0} not found.".format(user_param))
+    return user
+
+
+def resolve_github_repo_param(repo_param):
+    """Resolve GitGub repository CLI input parameter (name or GitHub ID)."""
+    try:
+        if '/' in repo_param:
+            repo = Repository.query.filter_by(name=repo_param).one()
+        else:
+            try:
+                github_id = int(repo_param)
+            except ValueError:
+                raise Exception('GitHub repository parameter must be either'
+                                ' a name or GitHub ID')
+            repo = Repository.query.filter_by(github_id=github_id).one()
+    except NoResultFound:
+        raise Exception("Repository {0} not found.".format(repo_param))
+    return repo
+
+
+@github.group()
+def repo():
+    """Repository management commands."""
+
+
+@repo.command()
+@click.argument('user')
+@click.option('--sync', '-s', default=False, is_flag=True,
+              help='Sync the repository prior to listing.')
+@click.option('with_all', '--all', '-a', default=False, is_flag=True,
+              help='List all repositories available.')
+@with_appcontext
+def list(user, sync, with_all):
+    """List user's repositories hooks.
+
+    Lists repositories currently `enabled` by the user.
+    If `--all` flag is specified, lists full list of repositories from remote
+    account extra data. For best result `--all` should be used with `--sync`,
+    so that the GitHub information is fresh. Positional argument USER can be
+    either an email or user ID.
+
+    Examples:
+
+      repo list foo@bar.baz
+
+      repo list 12345 --sync --all
+    """
+    user = resolve_user_param(user)
+    gha = GitHubAPI(user_id=user.id)
+    if sync:
+        gha.sync(hooks=True, async_hooks=False)  # Sync hooks asynchronously
+        db.session.commit()
+    if with_all:
+        repos = gha.account.extra_data['repos']
+        click.echo("User has {0} repositories in total.".format(len(repos)))
+        for gid, repo in repos.items():
+            click.echo(' {name}:{gid}'.format(name=repo['full_name'], gid=gid))
+
+    repos = Repository.query.filter_by(user_id=user.id)
+    click.echo("User has {0} enabled repositories.".format(repos.count()))
+    for r in repos:
+        click.echo(" {0}".format(r))
+
+
+def move_repository(repo, gha_old_user, gha_new_user):
+    """Transfer repository model and hook from one user to another."""
+    gha_old_user.remove_hook(repo.github_id, repo.name)
+    gha_new_user.create_hook(repo.github_id, repo.name)
+    db.session.commit()
+
+
+@repo.command()
+@click.argument('user')
+@click.option('repos', '--repo', '-r', help='GitHub Repository (name or ID)',
+              multiple=True)
+@click.option('all_of', '--all-of-user', '-A',
+              help="Change ownership of ALL current user's repositories.")
+@click.option('--yes-i-know', is_flag=True, default=False,
```

Comment:
The prompts are specific to other arguments provided, e.g.: "Creating a hook for X and Y. Continue?" I am not sure how easy it would be to have that with the option callback.